sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -378,11 +378,18 @@ object GpuCast extends Arm {
       // ansi cast from larger-than-long integral-like types, to long
       case (dt: DecimalType, LongType) if ansiMode =>
         // This is a work around for https://github.com/rapidsai/cudf/issues/9282
-        val min = BigDecimal(Long.MinValue)
+        val bigDecimalMin = BigDecimal(Long.MinValue)
             .setScale(dt.scale, BigDecimal.RoundingMode.DOWN).bigDecimal
-        val max = BigDecimal(Long.MaxValue)
+        val bigDecimalMax = BigDecimal(Long.MaxValue)
             .setScale(dt.scale, BigDecimal.RoundingMode.DOWN).bigDecimal
-        assertValuesInRange(input, Scalar.fromDecimal(min), Scalar.fromDecimal(max))
+        withResource(input.min()) { min =>
+          withResource(input.max()) { max =>
```

Comment:
`min` or `max` should be just a number, right? So we won't need to wrap them.

Reply:
This is not just a number, it is a `Scalar` object (https://github.com/rapidsai/cudf/blob/branch-22.08/java/src/main/java/ai/rapids/cudf/Scalar.java#L35), so it has to be closed.

Reply:
If it is scalar then we may want to call `isValid` to check the min and max values.